packages/teams-js/src/internal/utils.ts

```diff
@@ -39,6 +40,13 @@ function validateHostAgainstPattern(pattern: string, host: string): boolean {
 
 const validateOriginLogger = getLogger('validateOrigin');
 
+function validateOriginHandler(messageOrigin: URL): Promise<boolean> {
+  return new Promise<boolean>((resolve) => {
```

Comment:
>new Promise<boolean> [](http://example.com/codeflow?start=9&length=20)

Is double wrapping the `Promise` necessary here? Can this function just return the `Promise` returned from `sendAndUnwrap`?